### PR TITLE
[15.0][ADD] partner_name_hide_parent : New Module

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
This module does the following:

- Adds the ability to hide parent's name from the presentation of the partners' names according to the settings of the partners.
- The function provided by this module is needed when you have to set up a child partner (e.g. a branch office of a company) as an 'individual' due to how Odoo recognizes the commercial partner (i.e. if a child partner is a 'company', the partner becomes the commercial partner itself), and you do not want to show the parent's name in QWeb report or website where the partner is referenced (if the child partner is an individual, Odoo by default shows the parent's name preceding the child's name)

@qrtl

### CODE REVIEW

File: partner_name_hide_parent/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    _inherit = "res.partner"
+
+    hide_parent = fields.Boolean(
+        help="If selected, the parent's name will not be included in the "
+        "display name of self."
+    )
+
+    # Override the standard method
+    def _get_contact_name(self, partner, name):
+        # Following two lines are inserted. (QRTL)
```
Comment (yostashiro) [inline partner_name_hide_parent/models/res_partner.py]:
```suggestion
    def _get_contact_name(self, partner, name):
```

File: partner_name_hide_parent/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+This module does the following:
+
+* Adds the ability to hide parent's name from the presentation of the partners' names according to the settings of the partners.
+  The function provided by this module is needed when you have to set up a child partner (e.g. a branch office of a company) as an 'individual' due to how Odoo recognizes the commercial partner (i.e. if a child partner is a 'company', the partner becomes the commercial partner itself), and you do not want to show the parent's name in QWeb report or website where the partner is referenced (if the child partner is an individual, Odoo by default shows the parent's name preceding the child's name).
```
Comment (yostashiro) [inline partner_name_hide_parent/readme/DESCRIPTION.rst]:
```suggestion

The function provided by this module is needed when you have to set up a child partner
(e.g. a branch office of a company) as an 'individual' due to how Odoo recognizes the
commercial partner (i.e. if a child partner is a 'company', the partner becomes the
commercial partner itself), and you do not want to show the parent's name in QWeb report
or website where the partner is referenced (if the child partner is an individual, Odoo
by default shows the parent's name preceding the child's name).
```
Just to adjust the formatting.

File: partner_name_hide_parent/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2020-2022 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
```
Comment (yostashiro) [inline partner_name_hide_parent/__manifest__.py]:
```suggestion
# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
```
Can we switch to AGPL for this one.

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: partner_name_hide_parent/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,9 @@
+This module does the following:
+
+* Adds the ability to hide parent's name from the presentation of the partners' names according to the settings of the partners.
+* The function provided by this module is needed when you have to set up a child partner
+  (e.g. a branch office of a company) as an 'individual' due to how Odoo recognizes the
+  commercial partner (i.e. if a child partner is a 'company', the partner becomes the
+  commercial partner itself), and you do not want to show the parent's name in QWeb report
+  or website where the partner is referenced (if the child partner is an individual, Odoo
+  by default shows the parent's name preceding the child's name).
```
Comment (yostashiro) [inline partner_name_hide_parent/readme/DESCRIPTION.rst]:
```suggestion

The function provided by this module is needed when you have to set up a child partner
(e.g. a branch office of a company) as an 'individual' due to how Odoo recognizes the
commercial partner (i.e. if a child partner is a 'company', the partner becomes the
commercial partner itself), and you do not want to show the parent's name in QWeb report
or website where the partner is referenced (if the child partner is an individual, Odoo
by default shows the parent's name preceding the child's name).
```
This part explains the background and not the function, therefore it is not appropriate to be made part of what the module does.

**Review by yostashiro — CHANGES_REQUESTED** [review]

**Review by yostashiro — APPROVED** [review]
👍

**Review by rinaldifirdaus — APPROVED** [review]
Functional test in local environment and working well.

**Pull request merged**